kotlinx-coroutines-core/common/src/internal/LimitedDispatcher.kt

```diff
@@ -45,61 +44,37 @@ internal class LimitedDispatcher(
         return super.limitedParallelism(parallelism)
     }
 
-    override fun run() {
-        var fairnessCounter = 0
-        while (true) {
-            val task = queue.removeFirstOrNull()
-            if (task != null) {
-                try {
-                    task.run()
-                } catch (e: Throwable) {
-                    handleCoroutineException(EmptyCoroutineContext, e)
-                }
-                // 16 is our out-of-thin-air constant to emulate fairness. Used in JS dispatchers as well
-                if (++fairnessCounter >= 16 && dispatcher.isDispatchNeeded(this)) {
```

Comment:
Is it correct that this PR addresses the following issue in this snippet:

* `run` executes task (`:54`)
* Underlying dispatcher is closed
* `run` finishes, unluckily enough overflows `fairnessCounter`, goes to redispatch and crashes?

Reply:
That's one aspect of it, yes. Consider also the case where an external `dispatch` or `dispatchYield` happened (for example, from some callback), scheduled a task, waited for a millennia, during which this task and every other one got executed and the dispatcher got closed, and then proceeded to allocate a worker and run it.